core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -1896,13 +1896,11 @@ public void addView(View v) throws IOException {
     // even if we want to offer this atomic operation, CopyOnWriteArrayList
     // offers no such operation
     public void setViews(Collection<View> views) throws IOException {
-        BulkChange bc = new BulkChange(this);
-        try {
+        try (BulkChange bc = new BulkChange(this)) {
             this.views.clear();
             for (View v : views) {
                 addView(v);
             }
-        } finally {
```

Comment:
This doesn't look like it's the same, but it also looks like it might fix a bug…?

Reply:
Oh snap, it does fix a bug.

Reply:
Uhm... Indeed there is a bug in the original logic which may lead to partial view data on the disk. Not fun. Probably even worth a changelog entry.

Reply:
Well, why would there be a bulk change if you want to save partial data and never roll back?